staging/src/k8s.io/endpointslice/reconciler.go

```diff
@@ -261,6 +265,8 @@ func (r *Reconciler) reconcileByAddressType(logger klog.Logger, service *corev1.
 		Unchanged:   unchangedSlices(existingSlices, slicesToUpdate, slicesToDelete),
 	}
 
+	canUseTrafficDistribution := r.trafficDistributionEnabled && !hintsEnabled(service.Annotations)
+
 	if r.topologyCache != nil && hintsEnabled(service.Annotations) {
```

Comment:
just to be completely sure, in the future we remove this annotation functionality, right?
if that is the case we can add a comment explaining this and referencing the KEP